src/metabase/driver/sql/query_processor.clj

```diff
@@ -754,7 +754,21 @@
       (throw (ex-info "Summing intervals is not supported" {:args args})))
     (apply hx/call :+ (map (partial ->honeysql driver) args))))
 
-(defmethod ->honeysql [:sql :-] [driver [_ & args]] (apply hx/call :- (map (partial ->honeysql driver) args)))
+(defmethod ->honeysql [:sql :-]
+  [driver [_ & [first-arg & other-args :as args]]]
+  (assert (not (interval? first-arg))
+          "Interval as first argrument to subtraction is not allowed.")
+  (if (interval? (first other-args))
+    (do
+      (assert (every? interval? other-args)
```

Comment:
I'm a fan of "rather safe than sorry" attitude, but I have a feeling this assertion could unnecessary. Thanks for your opinion in advance :).

Reply:
See below. It's good to check these conditions, but they should be `Exception`s rather than `AssertionError`s.